content/blog/2024/jupyterhub-binderhub-gesis/index.md

```diff
@@ -0,0 +1,126 @@
+3. Talk to a JupyterHub instance to launch a user server with the built docker image, and redirect the user to this.
+
+(2) is really the *core* feature of BinderHub, and we settled on figuring out how to make that available to JupyterHub users. It was really important to us that this was also done in a way that can be sustainably used by *everyone*, not just 2i2c. This blog post discusses the various improvements to the broad ecosystem of projects in the Jupyter ecosystem to get this done.
+
+## Demo
+
+But first, a very quick demo of how this looks like right now now!
+
+<!-- generated from original .mov screen recording with `ffmpeg -i screencast.mov -c:v libx264 screencast.mp4` -->
+<video src="./screencast.mp4"></video>
+
+This is very much a work in progress, but the basic flow can be seen clearly. Users see this after they login to the JupyterHub. They can specify the two primary things that determine the server they start:
+
+1. The resources allocated (RAM, CPU and maybe GPU)
+2. The environment (docker image) used, which can be specified in one of 3 ways:
+
+    a. A pre-selected list of environments (docker images), provided by the administrators who set up this JupyterHub
+    b. A blank text box where you can enter any publicly available docker image they want
+    c. A mybinder.org style way to specify a GitHub repository, which will be then dynamically built into a docker image for the user!
+
+So what all did we need to do to accomplish this, in a way that's very upstream friendly and usable by everyone (and not just 2i2c)?
+
+## A Standalone `binderhub-service` helm chart
+
+The default upstream [BinderHub helm-chart](https://github.com/jupyterhub/binderhub/tree/main/helm-chart) *includes* a JupyterHub as a dependency, and configures itself to be used primarily in a manner similar to [mybinder.org](https://mybinder.org). As the person who helped make that choice early on, I can tell you why it was made - for convenience! And it *was* very convenient, as it allowed us to get mybinder.org going fast. However, it makes it difficult to install a BinderHub service *alongside* an existing JupyterHub. To this end, we have created a standalone [BinderHub helm chart](https://github.com/2i2c-org/binderhub-service/), designed to be installed *alongside* an existing JupyterHub, so we can use it *purely* to build images. This allows the BinderHub instance to be used as a [JupyterHub Service](https://jupyterhub.readthedocs.io/en/stable/reference/services.html), which is what we want.
+
+As part of this, we also added a way for BinderHub to run in [API only mode](https://github.com/jupyterhub/binderhub/pull/1647), so we can fully turn off the UI *and* launching ability of BinderHub, and use it purely for its building API.
```

Comment:
What are the usage scenario(s) for `API only mode`? I think a comment or two in this direction will strengthen the post.

Reply:
Added!